### PR TITLE
Adopt Flutter.xcframework in tool

### LINKED ISSUE #60109: Distribute engine as Flutter.xcframework
Xcode 11 (minimum required version in Flutter) introduced the .xcframework for distributing frameworks for multiple platforms and architectures. At build time, the correct version of the framework is linked on.  An XCFramework is already being used for add-to-app with the `flutter build ios-framework --xcframework` flag.

As Apple is increasing the number of architectures available for each platform (Catalyst as iOS apps on macOS, arm64 macOS) having one bundle containing all available architectures would reduce logic for the tool.  Particularly, no more re-copying the framework into the project when platforms architecture changes (still needs to be done per build mode), and no more `lipo` to remove unused architectures.

As an additional bonus, Flutter.xcframework can be distributed as a Swift Package [as of Xcode 12](https://developer.apple.com/documentation/swift_packages/distributing_binary_frameworks_as_swift_packages).  See https://github.com/flutter/flutter/issues/33850#issuecomment-647758430

With https://github.com/flutter/flutter/pull/51453 iOS linking and embedding logic has moved into the tool, so there won't be migration needed for existing projects.

- [x] Update LUCI recipes to create a Flutter.xcframework instead of a Flutter.framework.  Still need them separated for Debug, Profile, and Release.
- [ ] Flutter.podspec would need to change to `s.vendored_frameworks = 'Flutter.xcframework'`
- [ ] Minimum required CocoaPods version in the tool would become [1.9](https://blog.cocoapods.org/CocoaPods-1.9.0-beta/) to handle that vendored xcframework. https://github.com/flutter/flutter/pull/71170

### PR DESCRIPTION
**Pull request by jmagman**
## Description

XCFrameworks are bundles introduced in Xcode 11 that contain directories broken down by SDK (ex. iPhone, iPhone simulator, macOS, Catalyst macOS, etc) and architecture, plus an Info.plist with some metadata.

Flutter iOS engine artifacts [now contain an unzipped Flutter.xcframework](https://flutter-review.googlesource.com/c/recipes/+/9020).  (It also contains the old Flutter.framework.zip to support internal tooling, to be removed  with b/172736803).
```
./Flutter.xcframework/ios-armv7_arm64/Flutter.framework
./Flutter.xcframework/ios-x86_64-simulator/Flutter.framework
./Flutter.xcframework/Info.plist
./Flutter.framework
```

At some point, [ARM simulator bits](https://github.com/flutter/flutter/issues/64502) will be added on the engine side, and it _should_  Just Work™ on the tool side.

Adopt Flutter.xcframework bundle in tool.  Also adopt in add-to-app modules and host apps.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/60109
Part of https://github.com/flutter/flutter/issues/60118 via https://github.com/flutter/flutter/issues/64502
Part of https://github.com/flutter/flutter/issues/66474
Part of https://github.com/flutter/flutter/issues/33850 for add-to-app
Unblocks part of https://github.com/flutter/flutter/issues/70413

Related:
https://github.com/flutter/flutter/pull/69809
https://github.com/flutter/flutter/pull/70224
https://github.com/flutter/flutter/pull/71095
https://github.com/flutter/flutter/pull/71100
https://github.com/flutter/flutter/pull/71103
https://github.com/flutter/flutter/pull/71113

Engine:
https://github.com/flutter/engine/pull/22506
https://github.com/flutter/engine/pull/22664
https://flutter-review.googlesource.com/c/recipes/+/9020

## Tests

Most of this is being exercised by existing build tests.
- Added artifacts_test
- Updated `build_ios_framework_module_test`:
    - The engine builder is running Xcode 11, so creates a `ios-armv7_arm64` instead of a `ios-arm64_armv7` folder.  Support both names so an engine builder update doesn't require a coordinated framework roll.
    - The add-to-app release version of Flutter.xcframework now includes the debug simulator, matching existing release `Flutter.framework` behavior.

module_test_ios failure will be fixed by https://github.com/flutter/flutter/pull/71525
build_ios_framework_module_test will be fixed by https://github.com/flutter/flutter/pull/71531

### CODE REVIEW

**Comment by jmagman** [comment]
Blocked on an engine roll that contains `Flutter.xcframework` artifacts.

File: dev/devicelab/lib/framework/ios.dart
Comment (jmagman) [inline dev/devicelab/lib/framework/ios.dart:26]: Now release add-to-app fat `Flutter.framework` contains debug x86 so it can be built on the simulator.  `otool -l` doesn't show the bitcode section for a fat framework if any slices don't contain bitcode.  Specify `arm64` only since that's the part that actually gets shipped and requires bitcode.

**Comment by jmagman** [comment]
module_test_ios failure will be fixed by https://github.com/flutter/flutter/pull/71525

**Comment by jmagman** [comment]
build_ios_framework_module_test will be fixed by https://github.com/flutter/flutter/pull/71531

File: packages/flutter_tools/lib/src/ios/bitcode.dart
Comment (jmagman) [inline packages/flutter_tools/lib/src/ios/bitcode.dart:25]: This is now thrown in `getArtifactPath` > `_getIosEngineArtifactPath`

File: dev/devicelab/lib/framework/ios.dart
Comment (dnfield) [reply dev/devicelab/lib/framework/ios.dart:26]: What about armv7?

**Review by dnfield — APPROVED** [review]
LGTM once tests are passing, modulo questiona bout whether we should validate bitcode on arm7 as well as 64.

File: dev/devicelab/lib/framework/ios.dart
Comment (jmagman) [reply dev/devicelab/lib/framework/ios.dart:26]: This has mostly been useful as a sanity check to make sure a debug version wasn't incorrectly embedded, not that the builder is producing the right artifacts, so that was a shortcut.  I can check both if you prefer!

**Comment by jmagman** [comment]
Waiting on an engine roll to rebase and retest.

**Pull request merged**
